iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -658,6 +929,141 @@ TEST_F(PoshRuntime_test, ShutdownUnblocksBlockingPublisher)
     EXPECT_THAT(wasSampleSent.load(), Eq(true));
 }
 
+TEST_F(PoshRuntime_test, ShutdownUnblocksBlockingClient)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f67db1c5-8db9-4798-b73c-7175255c90fd");
+    // get publisher and subscriber
+    iox::capro::ServiceDescription serviceDescription{"don't", "stop", "me"};
+
+    iox::popo::ClientOptions clientOptions;
+    clientOptions.responseQueueCapacity = 10U;
+    clientOptions.responseQueueFullPolicy = iox::popo::QueueFullPolicy::BLOCK_PRODUCER;
+    clientOptions.serverTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+
+    iox::popo::ServerOptions serverOptions;
+    serverOptions.requestQueueCapacity = 1U;
+    serverOptions.requestQueueFullPolicy = iox::popo::QueueFullPolicy::BLOCK_PRODUCER;
+    serverOptions.clientTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+
+    iox::popo::UntypedClient client{serviceDescription, clientOptions};
+    iox::popo::UntypedServer server{serviceDescription, serverOptions};
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(300));
+
+    ASSERT_TRUE(server.hasClients());
+    ASSERT_THAT(client.getConnectionState(), Eq(iox::ConnectionState::CONNECTED));
+
+    auto threadSyncSemaphore = iox::posix::Semaphore::create(iox::posix::CreateUnnamedSingleProcessSemaphore, 0U);
+    std::atomic_bool wasRequestSent{false};
+
+    constexpr iox::units::Duration DEADLOCK_TIMEOUT{5_s};
+    Watchdog deadlockWatchdog{DEADLOCK_TIMEOUT};
+    deadlockWatchdog.watchAndActOnFailure([] { std::terminate(); });
+
+    // block in a separate thread
+    std::thread blockingServer([&] {
+        auto sendRequest = [&]() {
+            auto clientLoanResult = client.loan(sizeof(uint64_t), alignof(uint64_t));
+            ASSERT_FALSE(clientLoanResult.has_error());
+            client.send(clientLoanResult.value());
+        };
+
+        // send request till queue is full
+        for (uint64_t i = 0; i < serverOptions.requestQueueCapacity; ++i)
+        {
+            sendRequest();
+        }
+
+        // signal that an blocking sind is expected
+        ASSERT_FALSE(threadSyncSemaphore->post().has_error());
+        sendRequest();
+        wasRequestSent = true;
+    });
+
+    // wait some time to check if the client is blocked
+    constexpr int64_t SLEEP_IN_MS = 100;
```

Comment:
I don't really like seeing SI units encoded in the variable name instead of the type. You could think about moving this into line 986.